snapshot/restore/local_fs_output.go

```diff
@@ -299,7 +299,30 @@ func (o *FilesystemOutput) createDirectory(ctx context.Context, path string) err
 	}
 }
 
-func (o *FilesystemOutput) copyFileContent(ctx context.Context, targetPath string, f fs.File) error {
+func write(targetPath string, r fs.Reader) error {
+	f, err := os.Create(targetPath)
+	if err != nil {
+		return errors.Errorf("unable to create file %q: %v", targetPath, err)
+	}
+
+	// ensure we always close f. Note that this does not conflict with  the
+	// close below, as close is idempotent.
+	defer f.Close()
+
+	name := f.Name()
+
+	if _, err := io.Copy(f, r); err != nil {
```

Comment:
iocopy.JustCopy() will generally manage buffers better.

Reply:
iocopy.JustCopy: 96.9 seconds
io.Copy: 106 seconds
iocopy.JustCopy: 95 seconds
io.Copy: 122 seconds

after these 4 runs my results went a bit all over the place, maybe heat throttling or something, but iocopy.JustCopy definitely seems to perform much better